modules/mcagar/src/main/ts/ephox/mcagar/api/TinyHooks.ts

```diff
@@ -100,7 +106,7 @@ const bddSetupInShadowRoot = <T extends EditorType = EditorType>(settings: Recor
     };
   });
 
-  const hooks = setupHooks<T>(settings, setupModules, focusOnInit, () => Optional.from(editorDiv), () => teardown());
+  const hooks = setupHooks<T>(settings, setupModules, focusOnInit, () => Optional.some({ element: editorDiv, teardown }));
```

Comment:
I realised after looking at this code tonight, isn't going to work if ShadowDOM isn't supported, as it'll never setup the element. That's what this used `Optional.from` before as `editorDiv` could be undefined. So we'll probably need to change this to something like this maybe?
```suggestion
  const hooks = setupHooks<T>(settings, setupModules, focusOnInit, () => Optional.from(editorDiv).map((element) => ({ element, teardown })));
```

Reply:
There was a change in TinyHooks which made this simpler, can you take a look?

Reply:
https://github.com/tinymce/tinymce/pull/6418/files#diff-fc43fd9967dbe698b6e7271bb98988b8bb7bc7c273fcc4ad9d630a2a59a5652cR122